dask_geopandas/core.py

```diff
@@ -445,6 +445,60 @@ def to_parquet(self, path, *args, **kwargs):
 
         return to_parquet(self, path, *args, **kwargs)
 
+    def dissolve(self, by=None, aggfunc="first", split_out=1, **kwargs):
+        """Dissolve geometries within `groupby` into a single geometry.
+
+        Parameters
+        ----------
+        by : string, default None
+            Column whose values define groups to be dissolved. If None,
+            whole GeoDataFrame is considered a single group.
+        aggfunc : function,  string or dict, default "first"
+            Aggregation function for manipulation of data associated
+            with each group. Passed to dask `groupby.agg` method.
+            Note that ``aggfunc`` needs to be applicable to all column (i.e. ``"mean:``
+            cannot be used with string dtype). Select only required columns before
+            ``dissolve`` or pass a dictionary mapping ``aggfunc`` to each column
+            separately.
+        split_out : int, default 1
+            Number of partitions of the output
+
+        **kwargs
+            keyword arguments passed to `groupby`
+
+        Examples
+        --------
+        >>> ddf.dissolve("foo", split_out=12)
+
+        >>> ddf[["foo", "bar", "geometry"]].dissolve("foo", aggfunc="mean")
+
+        >>> ddf.dissolve("foo", aggfunc={"bar": "mean", "baz": "first"})
+
+        """
+        if by is None:
+            by = lambda x: 0
+            drop = [self.geometry.name]
+        else:
+            drop = [by, self.geometry.name]
+
+        def union(block):
+            merged_geom = block.unary_union
+            return merged_geom
+
+        merge_geometries = dd.Aggregation(
+            "merge_geometries", lambda s: s.agg(union), lambda s0: s0.agg(union)
+        )
+        if isinstance(aggfunc, dict):
+            data_agg = aggfunc
+        else:
+            data_agg = {col: aggfunc for col in self.columns.drop(drop)}
+        data_agg[self.geometry.name] = merge_geometries
+        aggregated = self.groupby(by=by, **kwargs).agg(
+            data_agg,
+            split_out=split_out,
+        )
+        return aggregated.set_crs(self.crs)
```

Comment:
Do we only want to set the crs of the `meta` here? Or is this actually needed for the results of dissolve in each partition? 
Because if it's only for meta, I think we should do this more targeted (eg `aggregated._meta.crs = ..`). And if not, that sounds as a bug in geopandas as well?

Reply:
Setting CRS of the `meta` is not enough, the computed df comes without a CRS. 

> And if not, that sounds as a bug in geopandas as well?

I don't think so. I GeoPandas we also set CRS manually in 
```py
aggregated_geometry = GeoDataFrame(g, geometry=self.geometry.name, crs=self.crs)
```

This `set_crs` is doing the same manual assignment.

Reply:
> I don't think so. I GeoPandas we also set CRS manually in

Ah, yes, I was for a moment thinking this was doing a geopandas' dissolve on each partition ... 

There is one difference though that with `set_crs` it's actually doing an extra copy of the data compared to `GeoDataFrame(..)`. But yeah, maybe something we can optimize later.